### DISCUSSION TITLE
"Note block" in Page Mode = Note in Edge Mode

### DISCUSSION THREAD

**Discussion by houshasen** [toeverything/AFFiNE#6989] [Ideas]
### Description

To improve the user experience and offer more flexibility in how users work, I propose integrating the ability to create notes in edge mode directly from page mode. This integration would allow for a smoother transition between the two modes.

**Creation**: I recommend adding a feature that enables users to create a "group of blocks," which could be referred to as a "note block", "group" or whatever in Page mode. This is essentially equivalent of a note in Edgeless mode. Users could form these groups by selecting multiple blocks and pressing Cmd-Shift-G in Page Mode. When switching to Edgeless mode, these groups would be recognized as individual notes. The reverse is true with notes in Edgeless mode is always represented as a note-block.

**Sorting**: Users should have the ability to intuitively drag and drop note blocks within the content zone, similar to table of contents in the Edgeless mode.

**Selection/Visualization:** Note blocks should be easily visible and selectable. A possible design option could include a second outer line left to the current block selection handle. This would allow users to quickly identify whether they are selecting a note block or an individual block.

### Use case

When taking notes, I typically start in Page mode, using headers and toggles to organize my thoughts. At this stage, I have a clear idea of the different sections or elements of my document, which would be better represented as individual notes in Edge mode. However, they initially appear as a single, large note in Edge Mode, requiring me to manually separate them. This feels like unnecessary work, especially since I've already organized them with headers.

The introduction of bidirectional note blocks could eliminate this extra step. Additionally, the reverse workflow of converting multiple notes back into Page mode and rearranging them directly in Page mode in the desired order for a long-form writing would be much more intuitive.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by pengx17** [comment]
We are designing a more seamless page/edgeless experience right now. I think @hwangdev97 can share some thoughts here.

**Comment by hwangdev97** [comment]
Your idea is excellent. We will continue to optimize this document and whiteboard integration feature.

The first step is to add the functionality to convert note directly into a doc on the edgeless in the next version.

![shot-2024-05-16-11 12 29@2x](https://github.com/toeverything/AFFiNE/assets/24713927/e11fe6b1-422c-4a00-aab4-0675f4bdd815)
👆 just look like this.

We have essentially clarified our future product direction and will continue along this path to complete our product. This is related to the Note Focus View you mentioned #6954  , but we need some more time to polish the details.

In the scenario you described, we will also add quicker ways to split content, allowing documents to be more easily separated onto the whiteboard in the future.

**Comment by houshasen** [comment]
The fundamental design and philosophy of AFFiNE are truly fascinating. I'm very excited about the app's potential and future.
